### PR TITLE
chore(ci): update layer ARN docs and create PR during release

### PR DESCRIPTION
**Pull request by heitorlessa**
**Issue number:** https://github.com/awslabs/aws-lambda-powertools-python/issues/2210

**DEPENDENCY** #2239

## Summary
Use new create-pr action introduced in https://github.com/awslabs/aws-lambda-powertools-python/pull/2238 to create a PR with latest Layer ARNs in the docs instead of pushing directly to develop as part of the release.

### Changes

> Please provide a summary of what's being changed

- [x] Propagate and scope permissions from `release.yml` workflow
- [x] Remove docs publishing responsibility from `reusable_deploy_v2_layer_stack.yml`
- [x] Split `release_docs` job in `publish_v2_layer.yml` workflow to: **(1)** create PR and, **(2)** deploy to latest released docs from PR temp branch
- [x] Document Layer Docs split (PR+Deploy from temp branch) and why
- [x] Remove now redundant `reusable_update_v2_layer_arn_docs.yml`
- [x] Update CHANGELOG with latest git tag available during release for the newly released docs 

### User experience

> Please share what the user experience looks like before and after this change

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] I have performed a self-review of this change
* [ ] Changes have been tested
* [x] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: .github/workflows/publish_v2_layer.yml [hunk 1]
```diff
@@ -87,6 +86,10 @@ jobs:
 
   beta:
     needs: build-layer
+    # lower privilege propagated from parent workflow (release.yml)
+    permissions:
+      id-token: write
+      contents: read
```
Comment (heitorlessa) [inline .github/workflows/publish_v2_layer.yml:92]:
```suggestion
      contents: read
      pages: write             # docs will be updated with latest Layer ARNs
      pull-requests: write # creation-action will create a PR with Layer ARN updates
```

File: .github/workflows/publish_v2_layer.yml [hunk 2]
```diff
@@ -97,6 +100,10 @@ jobs:
 
   prod:
     needs: beta
+    # lower privilege propagated from parent workflow (release.yml)
+    permissions:
+      id-token: write
+      contents: read
```
Comment (heitorlessa) [inline .github/workflows/publish_v2_layer.yml:108]:
```suggestion
      contents: read
      pages: write             # docs will be updated with latest Layer ARNs
      pull-requests: write # creation-action will create a PR with Layer ARN updates
```

File: .github/workflows/publish_v2_layer.yml [hunk 3]
```diff
@@ -125,10 +144,62 @@ jobs:
...
+          name: cdk-layer-stack
+          path: cdk-layer-stack/
+      - name: Replace layer versions in documentation
+        run: |
+          ls -la cdk-layer-stack/
+          ./layer/scripts/update_layer_arn.sh cdk-layer-stack
+      # NOTE: It felt unnecessary creating yet another PR to update changelog w/ latest tag
+      # since this is the only step in the release where we update docs from a temp branch
+      - name: Update changelog with latest tag
+        run: make changelog
```
Comment (heitorlessa) [inline .github/workflows/publish_v2_layer.yml:188]: oversight: we need to create a Git Tag as the release workflow runs before a Release Notes.

**Review by leandrodamascena — APPROVED** [review]
AMAZING WORK!!!

**Pull request merged**